packages/plugin-ext/src/plugin/type-converters.ts

```diff
@@ -835,10 +835,14 @@ export function fromTask(task: theia.Task): TaskDto | undefined {
     if ('detail' in task) {
         taskDto.detail = task.detail;
     }
-    if (typeof task.scope === 'object') {
-        taskDto.scope = task.scope.uri.toString();
-    } else if (typeof task.scope === 'number') {
-        taskDto.scope = task.scope;
+    if (task.scope) {
+        if (typeof task.scope === 'number') {
+            taskDto.scope = task.scope;
+        } else  {
+            taskDto.scope = task.scope.uri.toString();
+        }
+    } else {
+        taskDto.scope = types.TaskScope.Workspace;
```

Comment:
minor: we can avoid the nested loops

```suggestion
    if (typeof task.scope === 'number') {
        taskDto.scope = task.scope;
    } else if (task.scope !== undefined) {
        taskDto.scope = task.scope.uri.toString();
    } else {
        taskDto.scope = types.TaskScope.Workspace;
    }
```

Reply:
Thanks @vince-fugnitto, 
I have addressed your comments  and are now implemented in the latest commits, 
I have raised the following ticket to follow up the implementation of the proposed improvements: 
https://github.com/eclipse-theia/theia/issues/12455